MOM6/extra_sources/mom_cap_gtracer_flux.F90

```diff
@@ -0,0 +1,649 @@
+    !! ocean, defined by the field table, as well as a place holder of 
+    !! intermediate calculations, such as piston velocities, and parameters that 
+    !! impact the fluxes. The fields in this structure are never actually set, 
+    !! but the structure is used for initialisation of components and to spawn 
+    !! other structure whose fields are set.
+
+contains
+
+!> \brief Gas and tracer field initialization routine for running with MOM generic tracers.
+!! Copied and adapted slightly from
+!! https://github.com/NOAA-GFDL/FMScoupler/blob/77618869f48507c8629f28457cb701e25e1ea4fc/full/flux_exchange.F90#L626.
+subroutine gas_exchange_init (gas_fields_atm, gas_fields_ocn, gas_fluxes)
+  type(coupler_1d_bc_type), optional, pointer :: gas_fields_atm ! tracer fields in atm
+      !< Pointer to a structure containing atmospheric surface variables that
+      !! are used in the calculation of the atmosphere-ocean gas fluxes, as well
+      !! as parameters regulating these fluxes.
+  type(coupler_1d_bc_type), optional, pointer :: gas_fields_ocn ! tracer fields atop the ocean
+      !< Pointer to a structure containing ocean surface variables that are
+      !! used in the calculation of the atmosphere-ocean gas fluxes, as well as
+      !! parameters regulating these fluxes.
+  type(coupler_1d_bc_type), optional, pointer :: gas_fluxes ! tracer fluxes between the atm and ocean
+      !< Pointer to a structure for exchanging gas or tracer fluxes between the
+      !! atmosphere and ocean, defined by the field table, as well as a place holder
+      !! of intermediate calculations, such as piston velocities, and parameters
+      !! that impact the fluxes.
+
+  if (.not.gas_fluxes_initialized) then
+    call atmos_ocean_type_fluxes_init( )
+    call ocean_model_flux_init( )
+    call atmos_ocean_fluxes_init(ex_gas_fluxes, ex_gas_fields_atm, ex_gas_fields_ocn)
+    gas_fluxes_initialized = .true.
+  endif
+
+  if (present(gas_fields_atm)) gas_fields_atm => ex_gas_fields_atm
+  if (present(gas_fields_ocn)) gas_fields_ocn => ex_gas_fields_ocn
+  if (present(gas_fluxes)) gas_fluxes => ex_gas_fluxes
+
+end subroutine gas_exchange_init
+
+!> \brief Restore FMS coupler_bc_type state from ocean restart file
+! See https://github.com/NOAA-GFDL/FMScoupler/blob/main/full/coupler_main.F90#L1896
+subroutine gas_fields_restore(gas_fields, domain, directory)
+  type(coupler_2d_bc_type), intent(inout) :: gas_fields !< FMS coupler_bc_type to be registered for restarts
+  type(domain2D), intent(in)              :: domain     !< The FMS domain to use for this registration call
+  character(len=*), optional, intent(in)  :: directory  !< Directory containing the restart file
+
+  ! local variables
+  type(FmsNetcdfDomainFile_t), dimension(:), pointer :: ocn_bc_restart => NULL() !< Structures describing the restart files
+  integer                                            :: num_ocn_bc_restart = 0   !< The number of restart files to use
```

Comment:
I'm not sure if this is what you intended here, but in Fortran initialization on declaration implies the `save` attribute. This means the value of this variable will be stored between calls to this function.

Here it probably does not matter, has I assume the value will be overwritten when calling `coupler_type_register_restarts`, but it's always good to be aware of this. So I would suggest to remove the initialization.

Reply:
Thanks for the tip. I stole this from somewhere, so I'll check why it's being initialised there and remove if not necessary.

Reply:
If the code comes from somewhere else, just leave as it is. On the other hand, a comment about where the code comes from would be great.